server.go

```diff
@@ -396,6 +397,95 @@ func (s *server) GetPayment(ctx context.Context, in *breez.GetPaymentRequest) (*
 	return &breez.GetPaymentReply{PaymentError: sendResponse.PaymentError}, nil
 }
 
+func (s *server) GetSwapPayment(ctx context.Context, in *breez.GetSwapPaymentRequest) (*breez.GetSwapPaymentReply, error) {
+	redisConn := redisPool.Get()
+	m, err := redis.StringMap(redisConn.Do("HGETALL", "input-address:"+in.Address))
+	if err != nil {
+		log.Println("GetPayment error:", err)
+		return nil, status.Errorf(codes.Internal, "failed to retrieve address information")
+	}
+
+	// Ensure we didn't pay this invoice before
+	preImage := m["payment:PaymentPreimage"]
```

Comment:
The preImage should be taken from lightninglib payment. Currently there is not way to get a payment by hash but I think it is safe to not check this as LND will fail when trying to pay the same payment request twice.
My point here is we can avoid redis at all in this function and rely only on lightninglib.